File: card-api/src/test/java/com/jean202/cardmizer/api/demo/DemoScenarioControllerTest.java
package com.jean202.cardmizer.api.demo;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;
import java.util.Set;
import org.junit.jupiter.api.Test;

class DemoScenarioControllerTest {
    @Test
    void exposesScenarioTemplates() {
        RecommendationDemoScenarios scenarios = new RecommendationDemoScenarios(List.of(
                new RecommendationDemoScenario(
                        "demo-1",
                        "Demo Title",
                        "Demo Description",
                        "SAMSUNG_KPASS",
                        new DemoRecommendationRequest("2026-03", 20_000, "서울교통공사", null, Set.of()),
                        List.of()
                )
        ));
        DemoScenarioController controller = new DemoScenarioController(scenarios);

        List<DemoScenarioController.RecommendationDemoScenarioResponse> responses = controller.list();

        assertEquals(1, responses.size());
        assertEquals("demo-1", responses.get(0).id());
        assertEquals("2026-03", responses.get(0).request().spendingMonth());
        assertTrue(responses.get(0).expectedRecommendedCardId().equals("SAMSUNG_KPASS"));
    }
}

File: card-api/src/test/kotlin/com/jean202/cardmizer/api/demo/DemoScenarioControllerTest.kt
package com.jean202.cardmizer.api.demo

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class DemoScenarioControllerTest {
    @Test
    fun exposesScenarioTemplates() {
        val scenarios = RecommendationDemoScenarios(
            listOf(
                RecommendationDemoScenario(
                    id = "demo-1",
                    title = "Demo Title",
                    description = "Demo Description",
                    expectedRecommendedCardId = "SAMSUNG_KPASS",
                    request = DemoRecommendationRequest("2026-03", 20_000, "서울교통공사", null, emptySet()),
                    seedRecords = emptyList(),
                ),
            ),
        )
        val controller = DemoScenarioController(scenarios)

        val responses = controller.list()

        assertEquals(1, responses.size)
        assertEquals("demo-1", responses[0].id)
        assertEquals("2026-03", responses[0].request.spendingMonth)
        assertTrue(responses[0].expectedRecommendedCardId == "SAMSUNG_KPASS")
    }
}
